mock/py/mock-isolated-repo.py

```diff
@@ -0,0 +1,105 @@
+#! /usr/bin/python3
+
+"""
+Take the JSON provided by Mock, download corresponding RPMs, and put them into
+an RPM repository.
+"""
+
+# pylint: disable=invalid-name
+
+import argparse
+import concurrent.futures
+import json
+import logging
+import os
+import shutil
+import subprocess
+import sys
+
+import requests
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+
+def download_file(url, outputdir):
+    """
+    Download a single file (pool worker)
+    """
+    file_name = os.path.join(outputdir, os.path.basename(url))
+    log.info("Downloading %s", url)
+    try:
+        with requests.get(url, stream=True, timeout=30) as response:
+            if response.status_code != 200:
+                return False
+            with open(file_name, "wb") as fd:
+                shutil.copyfileobj(response.raw, fd)
```

Comment:
Reading https://docs.python-requests.org/en/latest/user/quickstart/#raw-response-content I wonder: should you use iter_content?

Reply:
It is possible to [work with `response.raw`](https://requests.readthedocs.io/en/latest/api/#requests.Response.raw), and then you can read the file using chunks with [copyfileobj]( https://docs.python.org/3/library/shutil.html#shutil.copyfileobj).